rct229/rulesets/ashrae9012019/section19/section19rule33.py

```diff
@@ -0,0 +1,76 @@
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_cooling import (
+    get_proposed_hvac_modeled_with_virtual_cooling,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_heating import (
+    get_proposed_hvac_modeled_with_virtual_heating,
+)
+from rct229.utils.assertions import getattr_
+
+
+class Section19Rule33(RuleDefinitionListIndexedBase):
+    """Rule 33 of ASHRAE 90.1-2019 Appendix G Section 19 (Envelope)"""
+
+    def __init__(self):
+        super(Section19Rule33, self).__init__(
+            rmrs_used=UserBaselineProposedVals(True, False, True),
+            each_rule=Section19Rule33.HVACRule(),
+            index_rmr="baseline",
+            id="19-33",
+            description="Where no heating and/or cooling system is to be installed, and a heating or cooling system is being simulated only to meet the requirements described in Section G3.1-10 HVAC Systems proposed column c and d, "
+            "heating and/or cooling system fans shall simulated to be cycled ON and OFF to meet heating and cooling loads during occupied hours in the proposed design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1-10 HVAC Systems proposed column c and d",
+            is_primary_rule=False,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmd_u = context.user
+        rmd_p = context.proposed
+        applicable_HVAC_systems_cooling_list_p = (
+            get_proposed_hvac_modeled_with_virtual_cooling(rmd_u, rmd_p)
+        )
+        applicable_HVAC_systems_heating_list_p = (
+            get_proposed_hvac_modeled_with_virtual_heating(rmd_u, rmd_p)
+        )
+
+        applicable_HVAC_systems_list_p = (
+            applicable_HVAC_systems_cooling_list_p
+            + applicable_HVAC_systems_heating_list_p
+        )
+        return {"applicable_HVAC_systems_list_p": applicable_HVAC_systems_list_p}
+
+    class HVACRule(PartialRuleDefinition):
+        def __init__(self):
+            super(Section19Rule33.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(True, False, True),
+            )
+
+        def applicability_check(self, context, calc_vals, data):
+            hvac_p = context.proposed
+            hvac_id_p = hvac_p["id"]
+            applicable_HVAC_systems_list_p = data["applicable_HVAC_systems_list_p"]
+
+            return hvac_id_p in applicable_HVAC_systems_list_p
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
```

Comment:
There is no Baseline model in this rule and we do not need a user model for the logic.

Reply:
Addressed